sdk/src/main/java/io/customer/sdk/queue/QueueRunRequest.kt

```diff
@@ -60,25 +64,47 @@ internal class QueueRunRequestImpl internal constructor(
                 val error = result.exceptionOrNull()
                 logger.debug("queue task $nextTaskStorageId run failed $error")
 
-                val customerIOError = error as? CustomerIOError
-                return if (customerIOError is CustomerIOError.HttpRequestsPaused) {
-                    // When HTTP requests are paused, don't increment metadata about the tasks to create inaccurate data
-                    logger.info("queue is quitting early because all HTTP requests are paused.")
-
-                    goToNextTask(emptyList(), totalNumberOfTasksToRun, null)
-                } else {
-                    val previousRunResults = nextTaskToRun.runResults
-                    val newRunResults = nextTaskToRun.runResults.copy(totalRuns = previousRunResults.totalRuns + 1)
-                    logger.debug("queue task $nextTaskStorageId, updating run history from: $previousRunResults to: $newRunResults")
-                    queueStorage.update(nextTaskStorageId, newRunResults)
-
-                    goToNextTask(inventory, totalNumberOfTasksToRun, nextTaskToRunInventoryItem)
+                return when (error as? CustomerIOError) {
+                    is CustomerIOError.HttpRequestsPaused -> {
+                        // When HTTP requests are paused, don't increment metadata about the tasks to create inaccurate data
+                        logger.info("queue is quitting early because all HTTP requests are paused.")
+
+                        goToNextTask(emptyList(), totalNumberOfTasksToRun, null)
+                    }
+                    is CustomerIOError.BadRequest400 -> {
+                        logger.debug("queue task $nextTaskStorageId failed with 400")
+
+                        logger.debug("queue deleting task $nextTaskStorageId because it will always fail")
+
+                        // check if its a groupStart task, if so delete the group otherwise just delete the task
+                        if (nextTaskToRunInventoryItem.groupStart == null) {
+                            queueStorage.delete(nextTaskStorageId)
+                        } else {
+                            queueStorage.deleteGroup(groupStartTask = nextTaskToRunInventoryItem.groupStart)
+                        }
+
+                        // since failed task isn't being saved, and the group is also deleted no need to update `lastFailedTask`
```

Comment:
🤔 I understand that there is no need to updating `lastFailedTask`, but is there a problem with updating `lastFailedTask`? 

I am worried about a side-effect happening from this code decision in the future. What if the logic of `goToNextTask` changes and passing `null` here as `lastFailedTask` causes a bug? 

I am wondering if maybe it's a better idea for us to use a non-null value here even if it doesn't make any difference to the code today. Maybe it will make a difference in the future? The task did indeed fail, so makes sense to use a non-null value?